src/policy/fees_util.cpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2024 The Bitcoin Core developers
+// Distributed under the MIT software license. See the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+
+#include <policy/fees_util.h>
+#include <primitives/transaction.h>
+
+TxAncestorsAndDescendants GetTxAncestorsAndDescendants(const std::vector<RemovedMempoolTransactionInfo>& transactions)
```

Comment:
+1 on separating this function out in a file and adding focussed unit tests for it, I feel this is generic enough to be used in several places later.